### ISSUE TITLE
Version 2.1.1 does not respect SSL settings

### ISSUE THREAD

**Issue by jakommo** [logstash-plugins/logstash-filter-elasticsearch#43]
- Version: 2.1.1 on LS 2.1.3
- Operating System: Ubuntu 16.04 
- Config File (if you have sensitive info, please remove it): see below
- Sample Data: -
- Steps to Reproduce:

config:

```
        elasticsearch {
            hosts => ["nuc01:9200"]
            index => "pfsense-*"
            ssl => true
            ca_file => "/path_to/CA.pem"
            query => "ip_address:%{[source_ip]} AND @timestamp:[%{[@timestamp]}||\-1h TO %{[@timestamp]}||-0s]"
            fields => [ ["sid", "sid"] ]
            add_tag => [ "enriched" ]
            user => "XXX"
            password => "XXX"
        }
```

This results in:

```
Failed to query elasticsearch for previous event {:index=>"pfsense-*", :query=>"ip_address:%{[source_ip]} AND @timestamp:[2016-08-19T16:57:04.238Z||\\-1h TO 2016-08-19T16:57:04.238Z||-0s]", :event=>#<LogStash::Event:0x50f2b06b @metadata_accessors=#<LogStash::Util::Accessors:0x29b78be3 @store={}, @lut={}>, @cancelled=false, @data={"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T16:57:04.238Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, @metadata={}, @accessors=#<LogStash::Util::Accessors:0x332e39b2 @store={"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T16:57:04.238Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, @lut={"type"=>[{"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T16:57:04.238Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, "type"], "host"=>[{"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T16:57:04.238Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, "host"], "[type]"=>[{"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T16:57:04.238Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, "type"], "[source_ip]"=>[{"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T16:57:04.238Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, "source_ip"], "[@timestamp]"=>[{"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T16:57:04.238Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, "@timestamp"]}>>, :error=>#<Faraday::ConnectionFailed>, :level=>:warn}
```

and on ES in:

```
[2016-08-19 18:57:04,286][WARN ][shield.transport.netty   ] [nuc01] received plaintext http traffic on a https channel, closing connection [id: 0xc1477410, /10.0.0.32:41610 => /10.0.0.10:9200]
```

It looks like `ssl => true` is not respected

Changing the config to:

```
hosts => ["https://nuc01:9200"]
```

the CA cert seems not to be used, resulting in:

LS:

```
Failed to query elasticsearch for previous event {:index=>"pfsense-*", :query=>"ip_address:%{[source_ip]} AND @timestamp:[2016-08-19T17:39:36.412Z||\\-1h TO 2016-08-19T17:39:36.412Z||-0s]", :event=>#<LogStash::Event:0x70e76c39 @metadata_accessors=#<LogStash::Util::Accessors:0x7cdfc184 @store={}, @lut={}>, @cancelled=false, @data={"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T17:39:36.412Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, @metadata={}, @accessors=#<LogStash::Util::Accessors:0x7bbc8a49 @store={"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T17:39:36.412Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, @lut={"type"=>[{"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T17:39:36.412Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, "type"], "host"=>[{"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T17:39:36.412Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, "host"], "[type]"=>[{"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T17:39:36.412Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, "type"], "[source_ip]"=>[{"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T17:39:36.412Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, "source_ip"], "[@timestamp]"=>[{"message"=>"10.0.0.10", "@version"=>"1", "@timestamp"=>"2016-08-19T17:39:36.412Z", "type"=>"paloaltoenriched", "host"=>"es-jre"}, "@timestamp"]}>>, :error=>#<Faraday::SSLError>, :level=>:warn}
```

and in ES:

```
[2016-08-19 19:39:37,252][WARN ][shield.transport.netty   ] [nuc01] Caught exception while handling client http traffic, closing connection [id: 0x05627715, /10.0.0.32:41868 => /10.0.0.10:9200]
javax.net.ssl.SSLException: Received fatal alert: certificate_unknown
...
```

**Comment by bosinm** [comment]
hey @jsvd,  wondering if you have had any chance to look at this

**Comment by jsvd** [comment]
@jakommo @bosinm with the patch in #44, setting `ssl => true` and `hosts => ["https://nuc01:9200"]` makes the `ca_file` option be respected

**Comment by JohnUSA** [comment]
Is this fixed? Even with patch #44 I still get the ConnectionFailed:  
`Failed to query elasticsearch for previous event {...}>>, :error=>#<Faraday::ConnectionFailed>, :level=>:warn}`

**Comment by jakommo** [comment]
@JohnUSA, did you upgrade the plugin to the latest version?

**Comment by JohnUSA** [comment]
Yeah, I did, made sure the code was there, restarted Logstash.

**Comment by jakommo** [comment]
@jsvd looks like https://github.com/logstash-plugins/logstash-filter-elasticsearch/pull/44 was not released yet. The latest plugin version is v3.0.2 which was built in July.

**Comment by jsvd** [comment]
this was released in 3.0.3 (current version is 3.2.0) so this can be closed.
